lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return starlark.Bool(re.MatchString(src)), nil
+}
+
+func find(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return starlark.String(re.FindString(src)), nil
+}
+
+func findAll(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return toTuple(re.FindAllString(src, max)), nil
+}
+
+func findSubmatches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return toTuple(re.FindStringSubmatch(src)), nil
+}
+
+func replaceAll(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src  string
+		repl starlark.Value
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &src, &repl); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	switch x := repl.(type) {
+	case starlark.Callable:
+		fn := func(matched string) string {
+			res, err := starlark.Call(thread, repl, starlark.Tuple{starlark.String(matched)}, nil)
+			if err != nil {
+				log.Printf("An error occured while executing the function: %s", err.Error())
+				return matched
+			}
+			resp, ok := res.(starlark.String)
+			if !ok {
+				log.Printf("A string is expected as return type of the function but was %s", res.Type())
+				return matched
+			}
+			return string(resp)
+		}
+		return starlark.String(re.ReplaceAllStringFunc(src, fn)), nil
+	case starlark.String:
+		return starlark.String(re.ReplaceAllString(src, string(x))), nil
+	}
+	return nil, fmt.Errorf("got %s, want a function or string", repl.Type())
+}
+
+func split(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return toTuple(re.Split(src, max)), nil
+}
+
+func builtinAttr(recv starlark.Value, name string, methods map[string]*starlark.Builtin) (starlark.Value, error) {
```

Comment:
Move these two functions up close to where they are referenced.

I wonder whether these functions (or something similar) should be part of the starlark package API.